lib/pavilion/test_config/resolver.py

```diff
@@ -440,6 +449,106 @@ def load_raw_configs(self, tests, host, modes):
 
         return picked_tests
 
+    def check_version_compatibility(self, test_suite, test_suite_cfg):
+        """Make sure test suites are compatible with current pavilion
+        version."""
+
+        # User may provide compatible_pav:versions in the following formats:
+        # Single Version, EX: 1.2.3
+        # Wild Card Version, EX: 1.2 or 1.2.*
+        #   Note: These have the same behavior
+        # Range of Versions, EX: 1.2.3-1.4.5, 1.2-3.4, 1.2-1.2.4, etc.
+
+        ABBRV_RE = re.compile(r'^\d+\.\d+$')
+        FULL_RE = re.compile(r'^\d+\.\d+\.\d+$')
+        WC_RE = re.compile(r'^\d+\.\d+\.\*$')
```

Comment:
you only need one regex.

Valid patterns ('n' is a sequence of digits):
- n
- n.*
- n.n
- n.n.*
- n.n.n

Groups and `|` patterns are your friend:
`r'a(bc|de)?'` matches `a`, `abc`, or `ade`